code/game/turfs/simulated/walls.dm

```diff
@@ -32,45 +35,26 @@
 		user << src.engraving
 
 /turf/simulated/wall/dismantle_wall(devastated = 0, explode = 0)
-	if(istype(src, /turf/simulated/wall/r_wall)) //Reinforced girder has deconstruction steps too. If no girder, drop ONE plasteel sheet AND rods
-		if(!devastated)
-			getFromPool(/obj/item/stack/sheet/plasteel, get_turf(src))
-			new /obj/structure/girder/reinforced(src)
-		else
-			getFromPool(/obj/item/stack/rods, get_turf(src), 2)
-			getFromPool(/obj/item/stack/sheet/plasteel, get_turf(src))
-	else if(istype(src,/turf/simulated/wall/cult))
-		if(!devastated)
-			var/obj/effect/decal/cleanable/blood/B = getFromPool(/obj/effect/decal/cleanable/blood, get_turf(src))
-			B.New(src)
-			new /obj/structure/cultgirder(src)
-		else
-			var/obj/effect/decal/cleanable/blood/B = getFromPool(/obj/effect/decal/cleanable/blood, get_turf(src))
-			B.New(src)
-			new /obj/effect/decal/remains/human(src)
-
+	if(mineral == "metal")

```

Comment:
I moved all the functions regarding r_walls to /turf/simulated/wall/r_wall/dismantle_wall, and the functions regarding cult walls to /turf/simulated/wall/cult/dismantle_wall.